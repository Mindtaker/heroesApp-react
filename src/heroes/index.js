// export * from './hooks';
export * from './components';
export * from './helpers';
export * from './pages';
export * from './routes/HeroesRoutes';
