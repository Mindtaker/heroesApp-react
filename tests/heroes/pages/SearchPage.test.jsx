import { MemoryRouter } from 'react-router-dom';
import { fireEvent, render, screen } from '@testing-library/react';

import { SearchPage } from '../../../src/heroes';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
  };
})

describe('SearchPage tests', () => {
  test('should show the default values', () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  test('should show Batman and the input element with the queryString value', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <SearchPage />
      </MemoryRouter>
    );

    const inputValue = screen.getByRole('textbox').value;
    expect(inputValue).toBe('batman');

    const imgSrc = screen.getByRole('img').src;
    expect(imgSrc).toContain('batman');

    const alertSearch = screen.getByTestId('alert-search');
    expect(alertSearch.className).toContain('d-none');

    const alertNotFound = screen.getByTestId('alert-not-found');
    expect(alertNotFound.className).toContain('d-none');
  });

  test('should show an error if hero is not found', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman123']}>
        <SearchPage />
      </MemoryRouter>
    );

    const inputValue = screen.getByRole('textbox').value;
    expect(inputValue).toBe('batman123');

    const alertSearch = screen.getByTestId('alert-search');
    expect(alertSearch.className).toContain('d-none');

    const alertNotFound = screen.getByTestId('alert-not-found');
    expect(alertNotFound.className).not.toContain('d-none');
  });

  test('should call the navigate to new page', () => {
    render(
      <MemoryRouter initialEntries={['/search']}>
        <SearchPage />
      </MemoryRouter>
    );

    const input = screen.getByRole('textbox');

    fireEvent.change(input, { target: { value: 'superman' } });

    const form = screen.getByTestId('search-form');
    fireEvent.submit(form);

    expect(mockNavigate).toHaveBeenCalledWith('?q=superman');
  });
});
