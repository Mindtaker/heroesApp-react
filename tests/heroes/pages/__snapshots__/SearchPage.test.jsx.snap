// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage tests should show the default values 1`] = `
<div>
  <h1>
    SearchPage
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        data-testid="search-form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero..."
          type="text"
          value=""
        />
        <button
          class="btn btn-dark mt-3"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-secondary animate__animated animate__fadeIn "
        data-testid="alert-search"
        role="alert"
      >
        Search a hero
      </div>
      <div
        class="alert alert-danger animate__animated animate__fadeIn d-none"
        data-testid="alert-not-found"
        role="alert"
      >
        Not hero with 
        <b />
      </div>
      <div
        class="d-flex flex-wrap gap-3"
      />
    </div>
  </div>
</div>
`;
